src/Orleans.Runtime/Catalog/ActivationData.cs

```diff
@@ -267,6 +267,8 @@ internal bool TryGetExtensionHandler(Type extensionType, out IGrainExtension res
 
         #endregion
 
+        public HashSet<GrainId> RunningRequestsSenders { get; } = new HashSet<GrainId>();
```

Comment:
For simplicity duplicate senders tracking wasn't implemented. Can be added in case of need.